moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -560,6 +560,84 @@ TEST_F(OneRobot, testPrintCurrentPositionWithJointLimits)
   state.printStatePositionsWithJointLimits(joint_model_group);
 }
 
+TEST_F(OneRobot, testInterpolation)
+{
+  moveit::core::RobotState state_a(robot_model_);
+
+  // Interpolate with itself
+  state_a.setToDefaultValues();
+  moveit::core::RobotState state_b(state_a);
+  moveit::core::RobotState interpolated_state(state_a);
+  for (size_t i = 0; i <= 10; ++i)
+  {
+    state_a.interpolate(state_b, static_cast<double>(i) / 10., interpolated_state,
+                        robot_model_->getJointModelGroup("base_from_base_to_e"));
+    EXPECT_TRUE(state_a.distance(state_b) < 1e-12);
+
+    for (const auto& link_name : robot_model_->getLinkModelNames())
+    {
+      EXPECT_FALSE(interpolated_state.getCollisionBodyTransform(link_name, 0).matrix().hasNaN());
```

Comment:
Do you mind leaving a reason here too?

Reply:
Done. I tried to add reasons everywhere, but some of them are probably redundant. Off the top of my head I can't imagine why some of these tests would fail. This particular check for NaNs was inspired by the original issue behind all this, which was because I was calling `interpolate` with a NaN interpolation parameter, which ended up littering the collision body transform with NaNs. Now, that particular error would be caught by the new check from #2664.